libs/wingsdk/src/target-sim/resource.ts

```diff
@@ -1,5 +1,202 @@
-import { ToSimulatorOutput } from "../simulator/simulator";
-import { IInflightHost, IResource } from "../std";
+import { mkdirSync, writeFileSync } from "fs";
+import { join, relative } from "path";
+import { Construct } from "constructs";
+import { SimResourceSchema } from "./schema-resources";
+import { bindSimulatorResource, makeSimulatorJsClient } from "./util";
+import { App, Lifting } from "../core";
+import { CaseConventions, ResourceNames } from "../shared/resource-names";
+import { ToSimulatorOutput } from "../simulator";
+import {
+  IInflight,
+  IInflightHost,
+  IResource as IStdResource,
+  Resource as StdResource,
+  Node,
+  Json,
+} from "../std";
+
+/**
+ * Contract that a resource backend must implement.
+ * @inflightinterface
```

Comment:
Why not just `@inflight`?

Reply:
It seemed like `@inflight` was already being used to mean several things, but I'm OK with overloading it if that's your preference. Here's how the `@inflight` docstring works now when added to a TypeScript interface (per your suggestion):

- `@inflight foo.IBar` - means this interface is a preflight interface, but the inflight "half" of it is defined in `foo.IBar`
- `@inflight` (without a name) - means this is an inflight interface

Reply:
Implemented in the PR

Reply:
I think they are all one and the same (concept)